raiden/network/proxies/token_network_registry.py

```diff
@@ -109,107 +139,296 @@ def add_token(
                 "the result of the precondition check is not precisely predictable."
             )
 
-        # check preconditions
+        if token_address == NULL_ADDRESS_BYTES:
+            raise InvalidTokenAddress("The call to register a token at 0x00..00 will fail.")
+
+        token_proxy = self.blockchain_service.token(token_address)
         try:
```

Comment:
Checklist of the preconditions:

- [ ] [`canCreateTokenNetwork` modified](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data/source/raiden/TokenNetworkRegistry.sol#L76)
- [x] [`require(token_to_token_networks[_token_address] == address(0x0));`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data/source/raiden/TokenNetworkRegistry.sol#L79), done by `already_registered = self.get_token_network`
- [x] [`require(_token_address != address(0x0));`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L228), done above by `if token_address == NULL_ADDRESS_BYTES:`
- [x] [`require(_secret_registry != address(0x0));`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L229), checked implicitly, if the registry exist this value is not zero.
- [x] [`require(_deprecation_executor != address(0x0));`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L230), checked implicitly, if the registry exist this value is not zero.
- [x] [`require(_chain_id > 0);`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L231), checked implicitly, if the registry exist this value is not zero.
- [x] [`require(_settlement_timeout_min > 0);`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L232), checked implicitly, if the registry exist this value is not zero.
- [x] [`require(_settlement_timeout_max > _settlement_timeout_min);`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L233), checked implicitly, if the registry exist this value is greater than the min.
- [x] [`require(contractExists(_token_address));`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L234),  checked implicitly, if the proxy exist the address was checked for code.
- [x] [require(contractExists(_secret_registry));``](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L235), checked implicitly, if the registry exist this value is greater than the min.
- [ ] [`require(_token_network_deposit_limit > 0);`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L237)
- [ ] [`require(_token_network_deposit_limit >= _channel_participant_deposit_limit);`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L238)
- [x] [`require(token.totalSupply() > 0);`](https://github.com/raiden-network/raiden-contracts/blob/94f1428d2fad4719efffb26be3d7731d7bb3f3da/raiden_contracts/data_0.24.0/source/raiden/TokenNetwork.sol#L248), done by `token_supply = token_proxy.total_supply(block_identifier=block_identifier)`.

Reply:
btw, the checks for:

```
require(_token_network_deposit_limit > 0);
require(_token_network_deposit_limit >= _channel_participant_deposit_limit);
```

can be done above, together with the check for the `NULL_ADDRESS_BYTES`.